### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by algonell**
Fix typos in documentation, comments, etc.

Via [codespell](https://github.com/codespell-project/codespell/).

### CODE REVIEW

File: spotbugs/src/main/java/edu/umd/cs/findbugs/SAXBugCollectionHandler.java [hunk 1]
```diff
@@ -300,7 +300,7 @@ public void startElement(String uri, String name, String qName, Attributes attri
                         try {
                             bugCollection.getProjectStats().setTimestamp(timestamp);
                         } catch (java.text.ParseException e) {
-                            throw new SAXException("Unparseable sequence number: '" + timestamp + "'", e);
+                            throw new SAXException("Unparsable sequence number: '" + timestamp + "'", e);
```
Comment (gtoison) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/SAXBugCollectionHandler.java:303]: That one is potentially debatable (or is it debateable? ;)

File: spotbugs/src/patches/bcel.diff
Comment (gtoison) [inline spotbugs/src/patches/bcel.diff:1]: This is a diff file and I don't think we want to change it (not sure why we're keeping it though).
Can you please revert the change on that file?

**Review by gtoison — CHANGES_REQUESTED** [review]

File: spotbugs/src/main/java/edu/umd/cs/findbugs/SAXBugCollectionHandler.java (hunk 1 above)
Comment (algonell) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/SAXBugCollectionHandler.java]: Absolutely, let's keep it as is.

**Review by gtoison — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -178,7 +178,7 @@ Currently the versioning policy of this project follows [Semantic Versioning v2.
 
 ### Added
 - New simple name-based AnnotationMatcher for exclude files (now bug annotations store the class java annotations in an attribute called `classAnnotationNames`). For example, use like <Match><Annotation name="org.immutables.value.Generated" /></Match> in an excludeFilter.xml to ignore classes generated by the Immutable framework. This ignores all class, method or field bugs in classes with that annotation.
-- Added the Common Weakness Enumeration (CWE) taxonomy to the Static Analysis Results Interchange Format (SARIF) report. The short and long description for the CWEs are retrived from a JSON file which is a slimmed down version of the official comprehensive CWE XML from MITRE. The JSON contains information about all CWEs. ([#2410](https://github.com/spotbugs/spotbugs/pull/2410)).
+- Added the Common Weakness Enumeration (CWE) taxonomy to the Static Analysis Results Interchange Format (SARIF) report. The short and long description for the CWEs are retrieved from a JSON file which is a slimmed down version of the official comprehensive CWE XML from MITRE. The JSON contains information about all CWEs. ([#2410](https://github.com/spotbugs/spotbugs/pull/2410)).
```
Comment (JuditKnoll) [inline CHANGELOG.md:181]: Updated this in GitHub's release page. At the other changes in the already released versions, there are only links to different versions of the CHANGELOG.md file.

**Review by JuditKnoll — CHANGES_REQUESTED** [review]

**Review by JuditKnoll — APPROVED** [review]

**Review by ThrawnCA — APPROVED** [review]
How satisfying :)

**Review by hazendaz — APPROVED** [review]
